src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -437,13 +451,62 @@ int kds_init_client(struct kds_sched *kds, struct kds_client *client)
 	return 0;
 }
 
+static inline void
+_kds_fini_client(struct kds_sched *kds, struct kds_client *client)
+{
+	struct kds_ctx_info info;
+	u32 bit;
+
+	client->abort = 1;
+
+	/* If the client was destroyed after all commands finished, set client
+	 * as abort and sleep doesn't impact anything.
+	 *
+	 * But if client was destroyed unexpected.
+	 * After set client as abort, CU subdevice should finished all unsubmitted
+	 * commands. The submitted command would finished as usual.
+	 * Sleep for 500ms is enought to wait all commands finished.
+	 *
+	 * Only when CU hang, the CU subdevice would be able to detect it and go
+	 * to bad status.
+	 * This client would know if CU is in bad status when close the
+	 * context.
+	 */
+	msleep(500);
```

Comment:
I feel this is odd too. :(
Since CU could be shared, so a client could not wait for a CU has no commands.
Unless the client count how many commands are not finished. Any better idea?

Reply:
fixed